### PR TITLE
Add Active Directory Domain Services metrics receiver

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
Adds the `active_directory_ds` metrics receiver.

Depends on #529

### CODE REVIEW

File: integration_test/third_party_apps_data/applications/active_directory_ds/metadata.yaml [hunk 1]
```diff
@@ -115,3 +124,105 @@ expected_logs:
...
     type: "[]string"
     description: "Dynamic string data that was used to construct the log message."
+expected_metrics:
+- type: workload.googleapis.com/active_directory.ds.bind.rate
+  value_type: DOUBLE
+  kind: GAUGE
+  monitored_resource: gce_instance
+  representative: true
+  labels:
+    type: server|client
```
Comment (jefferbrecht) [inline integration_test/third_party_apps_data/applications/active_directory_ds/metadata.yaml:134]: +1, thanks for populating the label regexes!

File: docs/active_directory_ds.md [hunk 2]
```diff
@@ -15,3 +15,21 @@ logging:
...
+metrics:
+  receivers:
+    active_directory_ds:
+      type: active_directory_ds
+  service:
+    pipelines:
+      active_directory_ds:
+        receivers:
+        - active_directory_ds
+```
```
[the remark this answers is not in the document]
Comment (BinaryFissionGames) [reply docs/active_directory_ds.md:68]: Wasn't aware that was still WIP - I've added in docs for metrics, and also for logs.

**Review by jefferbrecht — APPROVED** [review]
LGTM

File: docs/active_directory_ds.md [hunk 3]
```diff
@@ -15,3 +20,72 @@ logging:
...
+| active_directory.ds.replication.sync.request.count                | CUMULATIVE | {requests}        | result          | The number of sync requests made by the Directory Replication Agent.                                                     |
+| active_directory.ds.replication.object.rate                       | GAUGE      | {objects}/s       | direction       | The number of objects transmitted by the Directory Replication Agent per second.                                         |
+| active_directory.ds.replication.property.rate                     | GAUGE      | {properties}/s    | direction       | The number of properties transmitted by the Directory Replication Agent per second.                                      |
+| active_directory.ds.replication.value.rate                        | GAUGE      | {values}/s        | direction       | The number of values transmitted by the Directory Replication Agent per second.                                          |
```
Comment (jefferbrecht) [inline docs/active_directory_ds.md:79]: This metric also has a `type` label defined in metadata.

File: docs/active_directory_ds.md [hunk 4]
```diff
@@ -15,3 +20,72 @@ logging:
...
+| active_directory.ds.replication.operation.pending                 | GAUGE      | {operations}      | type            | The number of pending replication operations for the Directory Replication Agent.                                        |
```
Comment (jefferbrecht) [inline docs/active_directory_ds.md:80]: And this one doesn't -- I assume it just ended up on the wrong line and needs to be bumped up. That or the label in the metadata needs to be moved to this metric.

File: docs/active_directory_ds.md [hunk 5]
```diff
@@ -15,3 +20,72 @@ logging:
...
+| active_directory.ds.operation.rate                                | GAUGE      | {operations}/s    |                 | The number of operations performed per second.                                                                           |
```
Comment (jefferbrecht) [inline docs/active_directory_ds.md:81]: This metric also has a type label defined in metadata.

File: docs/active_directory_ds.md [hunk 6]
```diff
@@ -15,3 +20,72 @@ logging:
...
+| active_directory.ds.notification.queued                           | GAUGE      | {notifications}   |                 | The number of pending update notifications that have been queued to push to clients.                                     |
+| active_directory.ds.security_descriptor_propagations_event.queued | GAUGE      | {events}          |                 | The number of security descriptor propagation events that are queued for processing.                                     |
+| active_directory.ds.suboperation.rate                             | GAUGE      | {suboperations}/s | type            | The rate of sub-operations performed.                                                                                    |
+| active_directory.ds.bind.rate                                     | GAUGE      | {binds}/s         |                 | The number of binds per second serviced by this domain controller.                                                       |
```
Comment (jefferbrecht) [inline docs/active_directory_ds.md:86]: This metric also has a type label defined in metadata.

File: integration_test/third_party_apps_data/applications/active_directory_ds/metadata.yaml [hunk 7]
```diff
@@ -1,5 +1,6 @@
 short_name: AD DS
 long_name: Active Directory Domain Services
+supported_app_version: ["windows-server-2012-r2", "windows-server-2016", "windows-server-2019", "windows-server-2022"]
```
Comment (jefferbrecht) [inline integration_test/third_party_apps_data/applications/active_directory_ds/metadata.yaml:3]: While 2022 was included in the scoping doc, we haven't officially added support for 2022 to Ops Agent yet, so we should probably remove that from the list for now. We can add it back in once we add 2022 support.

**Comment by BinaryFissionGames** [comment]
Previous kokoro run, looks like active directory passes but iis/mssql fail due to metadata missing supported_versions:
```
--- FAIL: TestThirdPartyApps (0.45s)
    --- FAIL: TestThirdPartyApps/windows-2012-r2/iis (249.89s)
    --- FAIL: TestThirdPartyApps/windows-2019/iis (317.38s)
    --- FAIL: TestThirdPartyApps/sql-std-2019-win-2019/mssql (401.21s)
    --- FAIL: TestThirdPartyApps/sql-std-2019-win-2019/iis (402.06s)
    --- PASS: TestThirdPartyApps/windows-2012-r2/active_directory_ds (699.66s)
    --- PASS: TestThirdPartyApps/windows-2019/active_directory_ds (805.84s)
    --- PASS: TestThirdPartyApps/sql-std-2019-win-2019/active_directory_ds (886.72s)
```

**Comment by jefferbrecht** [comment]
The IIS and MSSQL PRs are adding supported_versions so we can ignore that for this PR.

**Pull request merged**